model-optimizer/extensions/middle/SliceConverter.py

```diff
@@ -16,18 +16,45 @@
 
 import numpy as np
 
+from extensions.ops.Cast import Cast
+from extensions.ops.gather import Gather
+from mo.front.caffe.extractors.utils import get_canonical_axis_index
 from mo.front.common.partial_infer.utils import int64_array
+from mo.front.tf.graph_utils import create_op_with_const_inputs
 from mo.graph.graph import Graph, rename_nodes
+from mo.graph.port import Port
 from mo.middle.replacement import MiddleReplacementPattern
+from mo.ops.clamp import Clamp
+from mo.ops.concat import Concat
 from mo.ops.const import Const
 from mo.ops.strided_slice import StridedSlice
-from mo.utils.error import Error
 
 
-def convert_negative_indices(indices: np.array, shape: np.array):
-    for ind, value in enumerate(indices):
-        if value < 0:
-            indices[ind] += shape[ind]
+def create_ss_interval_border(graph: Graph, slice_border_port: Port, shape, axes, node_name):
+    # the value for 'ends' might be maximum/minimum possible value of int64. This
+    # value must be converted to maximum/minimum of int32 because such big values do not fit into the int32 which is
+    # supported by the StridedSlice layer
+    clamp = create_op_with_const_inputs(graph, Clamp, port_value_dict={1: np.iinfo(np.int32).min,
+                                                                       2: np.iinfo(np.int32).max},
+                                        op_attrs=dict(name=node_name + '/Clamp'))
```

Comment:
nit: looks more readable to me. here and later
```suggestion
    clamp = create_op_with_const_inputs(
        graph, Clamp, port_value_dict={1: np.iinfo(np.int32).min, 2: np.iinfo(np.int32).max}, op_attrs=dict(name=node_name + '/Clamp'))
```